File: Backend/src/main/java/hadi/springSecurity/models/entities/Post.java
package hadi.springSecurity.models.entities;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;


@Entity
@Table(name = "Posts")
public class Post extends Content
{
	@OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
	private List<Comment> comments;
	private String underscoredTitle;
	public Post()
	{
	}

	public Post(String title, String text, String markdown, UserProfile profile)
	{
		super(title, text, markdown, profile);
		this.underscoredTitle = title.replace(" ", "_");
	}

	public List<Comment> getComments()
	{
		return comments;
	}

	public void setComments(List<Comment> comments)
	{
		this.comments = comments;
	}
	
	public boolean addComment(Comment comment)
	{
		if(comments.contains(comment))
			return false;
		comments.add(comment);
		return true;
	}
	
	public boolean deleteComment(Comment comment)
	{
		if(comments.contains(comment))
		{
			comments.remove(comment);
			return true;
		}
		return false;
	}

	public String getUnderscoredTitle()
	{
		return underscoredTitle;
	}

	public void setUnderscoredTitle(String underscoredTitle)
	{
		this.underscoredTitle = underscoredTitle;
	}
}

File: Frontend/src/app/models/entities/Post.ts
import { Comment } from "./Comment";
import { Content } from "./Content";
import { User } from "./User";
import { UserProfile } from "./UserProfile";

export class Post extends Content
{

    private _comments: Comment[];
    private _underscoredTitle: string;
    constructor(id: number, title: string, text: string, markdown: string, releaseDate: Date, lastEditDate: Date,
        hasBeenEdited: boolean, profile: UserProfile, authorName: string, comments: Comment[], underscoredTitle: string) 
    {
        super(id, title, text, markdown, releaseDate, lastEditDate, hasBeenEdited, profile, authorName);
        this._comments = comments;
        this._underscoredTitle = underscoredTitle;
    }
    static getEmptyPost(): Post
    {
        let date = new Date();
        return new Post(0, "Test", "This is my test text", "This is my **test** text", date, date, false, User.getDefaultUser().profile, "Guest", [], "Test");
    }

    /**
     * Getter comments
     * @return {Comment[]}
     */
    public get comments(): Comment[]
    {
        return this._comments;
    }

    /**
     * Setter comments
     * @param {Comment[]} value
     */
    public set comments(value: Comment[])
    {
        this._comments = value;
    }


    /**
     * Getter underscoredTitle
     * @return {string}
     */
    public get underscoredTitle(): string
    {
        if (this._underscoredTitle == null)
            return this.title.replace(" ", "/");
        return this._underscoredTitle;
    }

    /**
     * Setter underscoredTitle
     * @param {string} value
     */
    public set underscoredTitle(value: string)
    {
        this._underscoredTitle = value;
    }


}